opentelemetry-jaeger/src/exporter/runtime.rs

```diff
@@ -23,7 +23,7 @@ impl JaegerTraceRuntime for opentelemetry::runtime::Tokio {
     type Socket = tokio::net::UdpSocket;
 
     fn create_socket<T: ToSocketAddrs>(&self, host_port: T) -> thrift::Result<Self::Socket> {
-        let conn = std::net::UdpSocket::bind("0.0.0.0:0")?;
+        let conn = std::net::UdpSocket::bind(bind_addr(&host_port))?;
```

Comment:
This solution doesn't seem great, since it calls `to_socket_addrs()` twice (potentially resolving the host name twice). Instead, I think we should use a helper function more like this:

```rust
fn addrs_and_family(host_port: &impl ToSocketAddrs) -> Result<(Vec<SocketAddr>, SocketAddr), io::Error> {
    let mut addrs = host_port.to_socket_addrs()?.collect::<Vec<_>>();
    let family = match addrs.first() {
        Some(SocketAddr::V4(_)) | None => SocketAddr::from(([0, 0, 0, 0], 0)),
        Some(SocketAddr::V6(_)) => SocketAddr::from(([0, 0, 0, 0, 0, 0, 0, 0], 0)),
    };
    Ok((addrs, family))
}
```

We can then pass `&[SocketAddr]` to `connect()`.